File: Linkpoint/src/main/java/com/linkpoint/slproto/SLDefaultMessage.java
package com.linkpoint.slproto;

import com.linkpoint.slproto.messages.SLMessageHandler;
import java.nio.ByteBuffer;

public class SLDefaultMessage extends SLMessage {
    public int CalcPayloadSize() {
        return 0;
    }

    public void Handle(SLMessageHandler sLMessageHandler) {
        sLMessageHandler.DefaultMessageHandler(this);
    }

    public void PackPayload(ByteBuffer byteBuffer) {
    }

    public void UnpackPayload(ByteBuffer byteBuffer) {
    }
}

File: Linkpoint/src/main/java/com/linkpoint/slproto/SLDefaultMessage.kt
package com.linkpoint.slproto

import com.linkpoint.slproto.messages.SLMessageHandler
import java.nio.ByteBuffer

class SLDefaultMessage : SLMessage() {
    override fun CalcPayloadSize(): Int {
        return 0
    }

    override fun Handle(handler: SLMessageHandler) {
        handler.DefaultMessageHandler(this)
    }

    override fun PackPayload(buffer: ByteBuffer) {}

    override fun UnpackPayload(buffer: ByteBuffer) {}
}